### ISSUE TITLE
Too much silence in upmixing

### ISSUE THREAD

**Issue by CruelTott** [kinetiknz/cubeb#378]
When playing back 6 channel CUBEB_LAYOUT_3F2_LFE data to a soundcard set to 8 channels I only hear the audio from the first two channels.

I changed the part in cubeb_upmix that inserts silence from:
```
  for (long i = 0, o = 0; i < inframes; ++i, o += out_channels) {
    for (unsigned int j = 2; j < out_channels; ++j) {
      assert((unsigned long)o + j < out_len);
      out[o + j] = 0.0;
    }
  }
```
to:
```
  for (long i = 0, o = 0; i < inframes; ++i, o += out_channels) {
    for (unsigned int j = in_channels; j < out_channels; ++j) {
      assert((unsigned long)o + j < out_len);
      out[o + j] = 0.0;
    }
  }
```
which seems to have fixed it for me.

There is probably some more correct way to map the two rear channels from the 6 channel source to the 4 rear ones in the 8 channel output, but at least with this fix no audio gets lost.

**Comment by kinetiknz** [comment]
@ChunMinChang Do you have time to look at this?

**Comment by kinetiknz** [comment]
And thank you for the bug report @CruelTott 👍

**Comment by ChunMinChang** [comment]
Thanks for the report! I address this in #388.

**Comment by kinetiknz** [comment]
Closed by #388.
